File: go/handlers/metadata_handler.go
package handlers

import (
	"encoding/json"
	"net/http"

	"filtex-go-example/filters"
)

func MetadataHandler(w http.ResponseWriter, r *http.Request) {
	projectFilter, err := filters.ProjectFilter()
	if err != nil {
		http.Error(w, "Error generating Filtex instance", http.StatusInternalServerError)
		return
	}

	response, err := projectFilter.Metadata()
	if err != nil {
		http.Error(w, "Error getting Filtex metadata", http.StatusInternalServerError)
		return
	}

	jsonResponse, err := json.Marshal(response)
	if err != nil {
		http.Error(w, "Error encoding JSON response", http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")

	w.Write(jsonResponse)
}

File: js/handlers/metadataHandler.js
const { ProjectFilter } = require('../filters/projectFilter');

class MetadataHandler {
    async handleAsync(req, res) {
        try {
            const response = ProjectFilter.metadata();

            res.status(200).json(response);
        } catch (ex) {
            res.status(500).send(ex.message);
        }
    }
}

module.exports = { MetadataHandler };
